src/librustc_driver/driver.rs

```diff
@@ -934,6 +934,7 @@ pub fn phase_3_run_analysis_passes<'tcx, F, R>(sess: &'tcx Session,
     passes.push_pass(MIR_CONST, mir::transform::rustc_peek::SanityCheck);
 
     // What we need to run borrowck etc.
+    passes.push_pass(MIR_CONST, mir::transform::borrow_check::BorrowckMir);
     passes.push_pass(MIR_VALIDATED, mir::transform::qualify_consts::QualifyAndPromoteConstants);
     passes.push_pass(MIR_VALIDATED,
```

Comment:
Hmm, I see. I guess niko's (and perhaps others' ) expectation was that borrowck would be injected in between `MIR_VALIDATED` and `MIR_OPTIMIZED`; but this PR as written is running  its MIR borrowck right before the `qualify_consts` step, which does not match that expectation.

Still not 100% sure how we'll end up dealing with this. I'll add a FIXME regarding the conflict for now.